src/main/java/gregtech/api/cover/CoverSaveHandler.java

```diff
@@ -0,0 +1,159 @@
+package gregtech.api.cover;
+
+import gregtech.api.util.GTLog;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.BiConsumer;
+
+public final class CoverSaveHandler {
+
+    private CoverSaveHandler() {}
+
+    /**
+     * Write sync data for a CoverableView's covers
+     *
+     * @param buf           the buf to write to
+     * @param coverableView the CoverableView containing the covers
+     */
+    public static void writeInitialSyncData(@NotNull PacketBuffer buf, @NotNull CoverableView coverableView) {
+        Cover[] covers = new Cover[EnumFacing.VALUES.length];
+        int count = 0;
+        for (EnumFacing facing : EnumFacing.VALUES) {
+            Cover cover = coverableView.getCoverAtSide(facing);
+            if (cover != null) {
+                covers[count++] = cover;
+            }
+        }
+
+        // need to write count so the reader knows not to do anything for covers
+        buf.writeByte(count);
+        if (count == 0) return;
+
+        for (int i = 0; i < count; i++) {
+            Cover cover = covers[i];
+            buf.writeByte(cover.getAttachedSide().ordinal());
+            buf.writeVarInt(CoverDefinition.getNetworkIdForCover(cover.getDefinition()));
```

Comment:
You could probably shave a byte of by packing side and id into 1 int. The side is from 0 to 5 and can therefore be represented by 3 bits. So something like  `int packed = (side & 0b111) & (network << 3)`.
If you wanted to